### DISCUSSION TITLE
Using high-end IMU absolute rotation in the Bundle adjustment

### DISCUSSION THREAD

**Discussion by khshmt** [openMVG/openMVG#2293] [General]
Hi all,

I am seeking a technical opinion to go or not to go in what I am about to do which is trying to extend the reprojection error cost function to add more constraint depends on the absolute rotation coming from an industrial grade AHRS(e.g., Vectornav IMUs) as a reference for the camera estimated rotation, So in addition to the reporjection error we add also the error induced from the difference between the estimated camera rotation and the IMU rotation.

So I like to hear from you about the validity of this idea regardless of any technical hustle like IMU initialization or different reference frames for the camera and the IMU.

Thanks in advance.

**Comment by pmoulon** [comment]
For me I think what you could consider add those for global SfM as additional constraint for rotation averaging (by adding the relative rotations computed from your global imu rotations). 

Else using inu prior during SfM is hard since you would end up with residual of different orders (pixel vs angular) and maybe using pose graph optimization would work better for global imu priors.

**Reply by khshmt** [reply]
So what about using the relative rotations computed from the IMU as a constraint for rotation averaging but for Incremental SFM, what is the downfall here?

Thanks @pmoulon

**Reply by pmoulon** [reply]
If you want to use it for Incremental SfM, at the end you just need to perform translation estimation. So you could use Translation estimation from 2points with ransac -> https://github.com/openMVG/openMVG/blob/develop/src/openMVG/multiview/solver_translation_knownRotation_kernel.hpp#L31
